### DISCUSSION TITLE
Bluetooth SPP master role

### DISCUSSION THREAD

**Discussion by lyusupov** [earlephilhower/arduino-pico#1361] [Ideas]
[SerialBT.cpp](https://github.com/earlephilhower/arduino-pico/blob/master/libraries/SerialBT/src/SerialBT.cpp) does currently have implementation for a "salve role' only.

It would be nice to have support for the 'master' role as well _(like [Arduino Core for ESP32](https://github.com/espressif/arduino-esp32/tree/master/libraries/BluetoothSerial) does)_.This will give a chance to connect a couple of PicoWs to each other by means of the Bluetooth transport. 

Another issue is that ["PicoW Serial"](https://github.com/earlephilhower/arduino-pico/blob/master/libraries/SerialBT/src/SerialBT.cpp#L67) 'slave' id is hard-coded into the library.
A user may need to alter this name onto a custom one.

**Comment by George-Still-Learning** [comment]
I have a project that uses a HC-05 Bluetooth SPP module in the "slave" mode to Tx data in the form of serial text to my PC (GBU521 USB module) and Android tablet/phone.  Now that I have moved into the Arduino driven Pico W I would like to use the Pico's on board Bluetooth to Rx this data also, but I can't because as stated above it is "implementation for a "slave role' only".  There are many of these HC05/6's in use out there.

Is there any implementation or near future plans a for serial "master" mode?  Or is one of the existing Bluetooth functions able to do this?

Using the Arduino/earlephilhower/Pico W is allowing me to join a variety functions/items into one package.  Very Nice!

tnx
George
